nifi-nar-bundles/nifi-azure-bundle/nifi-azure-services-api/pom.xml

```diff
@@ -40,5 +40,9 @@
             <groupId>org.apache.nifi</groupId>
             <artifactId>nifi-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.azure</groupId>
+            <artifactId>azure-storage-blob</artifactId>
+        </dependency>
```

Comment:
Although this module has some other Azure libraries, it is best to avoid introducing third-party libraries in Controller Service API modules as much as possible. It appears the only reason is for the `BlobErrorCode` in `AzureStorageConflictResolutionStrategy`. Recommend removing this dependency and replacing the direct reference to `BLOB_ALREADY_EXISTS` with a regular string description.

Reply:
That's fair – fixed. Doesn't seem like something they'll go and change any day either.